app/controllers/pieces_controller.rb

```diff
@@ -0,0 +1,14 @@
+class PiecesController < ApplicationController
+	def show 
+		@piece = Piece.find(params[:id])
+	end
+
+	def update
+		@piece = Piece.find(params[:id])
+		@game = Game.find(params[:id])
+		if @piece.valid_move?([:x, :y]) && @game(piece_in_space?(x, y) == false)
```

Comment:
`@game(piece_in_space?(x, y) == false` is not necessary or valid code. The logic that deals with checking whether or not pieces are in the way exists in the `is_obstructed` method which should exist within the `move_to` method. In addition, the premise of causing the move to fail if a piece is in the destination square is invalid. Moving a piece to a square that has the opponent's piece on it is a valid move.